File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] a) {
        int n=a.length;
        for(int i=n-1;i>=0;i--)
        {
            if(a[i]!=9)
            {
                a[i]+=1;
                return a;
            }
            a[i]=0;
        }
        int[] b=new int[n+1];
        b[0]=1;
        return b;
    }
}

File: 0066-plus-one/0066-plus-one.c
int* plusOne(int* a, int n, int* r) 
{
    int *s=(int *)malloc((n+1)*sizeof(int));
    int sum=0,r1,c=1;
    for(int i=n-1;i>=0;i--)
    {
        sum=a[i]+c;
        s[i+1]=sum%10;
        c=sum/10;
    }
    if(c)
    {
        s[0]=c;
        *r=n+1; 
    }
    else{
        for(int i=0;i<n;i++)
        {
            s[i]=s[i+1];
        }
        *r=n;
    }
    return s;
}